src/patch-prototypes.js

```diff
@@ -0,0 +1,94 @@
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+*/
+
+import * as utils from './utils.js';
+import {EventTargetPatches} from './patches/EventTarget.js';
+import {NodePatches} from './patches/Node.js';
+import {SlotablePatches} from './patches/Slotable.js';
+import {ParentNodePatches, ParentNodeDocumentOrFragmentPatches} from './patches/ParentNode.js';
+import {ElementPatches} from './patches/Element.js';
+import {ElementOrShadowRootPatches} from './patches/ElementOrShadowRoot.js';
+import {HTMLElementPatches} from './patches/HTMLElement.js';
+import {SlotPatches} from './patches/Slot.js';
+import {DocumentOrFragmentPatches} from './patches/DocumentOrFragment.js';
+import {DocumentOrShadowRootPatches} from './patches/DocumentOrShadowRoot.js';
+import {DocumentPatches} from './patches/Document.js';
+import {WindowPatches} from './patches/Window.js';
+
+// Some browsers (IE/Edge) have non-standard HTMLElement accessors.
+const NonStandardHTMLElement = {};
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'parentElement')) {
+  Object.defineProperty(NonStandardHTMLElement, 'parentElement',
+    Object.getOwnPropertyDescriptor(NodePatches, 'parentElement'));
+}
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'contains')) {
+  Object.defineProperty(NonStandardHTMLElement, 'contains',
+    Object.getOwnPropertyDescriptor(NodePatches, 'contains'));
+}
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'children')) {
+  Object.defineProperty(NonStandardHTMLElement, 'children',
+    Object.getOwnPropertyDescriptor(ParentNodePatches, 'children'));
+}
+
+if (Object.getOwnPropertyDescriptor(utils.NativeHTMLElement.prototype, 'innerHTML')) {
+  Object.defineProperty(NonStandardHTMLElement, 'innerHTML',
+    Object.getOwnPropertyDescriptor(ElementOrShadowRootPatches, 'innerHTML'));
+}
+
+// setup patching
+const patchMap = {
+  EventTarget: [EventTargetPatches],
+  Node: [NodePatches, !window.EventTarget ? EventTargetPatches : null],
+  Text: [SlotablePatches],
+  Element: [ElementPatches, ParentNodePatches, SlotablePatches,
+    utils.settings.IS_IE ? null : ElementOrShadowRootPatches,
+    !window.HTMLSlotElement ? SlotPatches : null],
+  HTMLElement: [HTMLElementPatches, NonStandardHTMLElement],
+  HTMLSlotElement: [SlotPatches],
+  DocumentFragment: [ParentNodeDocumentOrFragmentPatches, DocumentOrFragmentPatches],
+  Document: [DocumentPatches, ParentNodeDocumentOrFragmentPatches, DocumentOrFragmentPatches, DocumentOrShadowRootPatches],
+  Window: [WindowPatches]
+}
+
+const getPatchPrototype = (name) => (name === 'HTMLElement') ?
+    utils.NativeHTMLElement.prototype :
+    window[name] && window[name].prototype;
+
+const onPatchMap = (fn) => {
```

Comment:
Can eliminate the callback, move that inside, and call `applyPatches` with a `prefix` argument.

Reply:
Done.